api/nodes/serializers.py

```diff
@@ -150,13 +151,17 @@ def create(self, validated_data):
         auth = Auth(user)
         node = self.context['view'].get_node()
         pointer_node = Node.load(validated_data['node']['_id'])
-        pointer = node.add_pointer(pointer_node, auth, save=True)
-        return pointer
+        if not pointer_node:
+            raise exceptions.ParseError('Node not found.')
+        try:
+            pointer = node.add_pointer(pointer_node, auth, save=True)
+            return pointer
+        except ValueError:
+            raise exceptions.ParseError('Pointer to node {} already in list'.format(pointer_node._id))
```

Comment:
FWIW, looks like there's also a `ValidationError` that returns a `400`: https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/exceptions.py#L57